examples/direct_fidelity_estimation.py

```diff
@@ -196,6 +199,23 @@ def _estimate_pauli_traces_general(qubits: List[cirq.Qid],
     return pauli_traces
 
 
+@dataclass
+class DFEIntermediateResult:
+    """
+    A container for the various debug and run data from calling the function
+    direct_fidelity_estimation(). This is useful when running a long-computation
+    on an actual computer, which is expensive. This way, runs can be more easily
+    debugged offline.
+    """
+    # If the circuit is Clifford, the list of Clifford basis states. Otherwise
+    # it is None.
+    clifford_state: Optional[cirq.CliffordState]
+    # The list of Pauli traces we can sample from.
+    pauli_traces: List[Dict[str, Any]]
```

Comment:
Using the `Any` type essentially disables type checking, so we shouldn't use it unless there is really no other way. Could you replace `Any` with more concrete types here and in `trial_results` below?

(This does have a lot of knock-on effects, but type annotations are worth the effort of putting them in.)

Reply:
OK. So let me sure I understand what it is that you are asking here.

I think the main issue is that the values of the dictionary have heterogeneous types, and thus having an annotation is difficult. I seems that it's possible with TypedDict:
https://stackoverflow.com/questions/48054521/indicating-multiple-value-in-a-dict-for-type-hints

But this seems to be available in Python 3.8:
https://www.python.org/dev/peps/pep-0589/

However, it seems that we are a version < 3.7
https://github.com/quantumlib/Cirq/blob/master/requirements.txt

Am I correct? I'm leaving this unresolved because I am happy to address the comment, it's just that I could use a little help when you have the time.
Am I correct?

Reply:
Actually, I though about it overnight and you had already given the answer: Use dataclasses, which I did. Sorry for being obtuse.

No more "Any" type.